Packs/CheckPointNDR/Integrations/CheckPointNDR/CheckPointNDR.py

```diff
@@ -0,0 +1,176 @@
+import urllib3
+import demistomock as demisto
+from CommonServerPython import *
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+NDR_URL = 'https://now.checkpoint.com'
+
+
+class Client(BaseClient):
+    def __init__(self, base_url: str, client_id: str, access_key: str, domain: str, verify: bool, proxy: bool):
+
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy)
+
+        self.client_id = client_id
+        self.access_key = access_key
+        self.domain = domain
+
+    def _login(self) -> None:
+        auth_data = {"selectedRealm": "ssl_vpn_Username_Password",
+                     "userName": self.client_id,
+                     "password": self.access_key}
+
+        if self._session.cookies:
+            self._session.cookies.clear()
+
+        res = self._http_request('POST', url_suffix='/Login/LoginAPI', data=auth_data, resp_type='response')
+
+        if res.status_code == 200:
+            demisto.debug(f"Log-in successful, new API endpoint is: {res.url}")
+            self._base_url = res.url
+        else:
+            raise DemistoException(f"Log-in failed: {str(res.status_code)}: {res.text}")
+
+    def _logout(self) -> None:
+        if self._session.cookies:
+            self._http_request('POST', url_suffix='/../Portal/SignOut', resp_type='response')
+
+    def _call_api(self,
+                  url_suffix: str,
+                  method: str = "GET",
+                  params: Optional[dict[str, Any]] = None,
+                  json_data: Optional[dict[str, Any]] = None) -> list[dict[str, Any]]:
+
+        res_json = self._http_request(
+            method,
+            url_suffix=f'/incidents/v1/{url_suffix}',
+            headers={'domain': self.domain},
+            params=params,
+            json_data=json_data
+        )
+
+        if res_json.get('status') and res_json.get('status') == "error":
+            raise DemistoException(f"API call failed: {res_json.get('message')}")
+        return res_json.get('objects')
+
+    def get_insights(self, startTS: int, max_fetch: int) -> tuple[list[dict[str, Any]], datetime]:
+        lastTS = startTS
+
+        self._login()
+        insights = self._call_api('insights', params={'updated': f"gt.{startTS}"})
+        demisto.debug(f"Fetched {len(insights)} NDR Insights, processing {min(len(insights), max_fetch)} of them...")
+
+        insights = sorted(insights, key=lambda x: x['updated'])[:max_fetch]
+        for insight in insights:
+            ids = ','.join(map(str, insight['events']))
+            insight['events'] = self._call_api('events', params={'id': ids})
+            demisto.debug(f"Fetched {len(insight['events'])} events of Insight {insight['id']}")
+        if(len(insights) > 0):
+            lastTS = insights[-1]['updated']
```

Comment:
In what order does it return from the API?
It seems that each insight['events'] has more than one event,
and each of them is a separate incident,
So we return more incidents than the max_fetch

Reply:
You're right, I'm handling this now in parse_insights